### PR TITLE
Use events model for pipeline and build pages

### PR DESCRIPTION
**Pull request by petey**
Uses the new events model to pull information about commit history and committers for pipeline and build pages

**Pipeline page**
screen shot 2016-11-08 at 9 20 33 pm

**Build Page**
screen shot 2016-11-08 at 9 20 54 pm

Replaces pipeline-build-* components with pipeline-event-*. Removes build-details component that was unused.

### CODE REVIEW

File: app/pipeline/model.js [hunk 1]
```diff
@@ -2,32 +2,15 @@ import DS from 'ember-data';
 import Ember from 'ember';
 
 export default DS.Model.extend({
-  scmUrl: DS.attr('string'),
-  configUrl: DS.attr('string', { defaultValue: '' }),
-  scmUri: DS.attr('string'),
-  createTime: DS.attr('date'),
   admins: DS.attr(),
-  secrets: DS.hasMany('secret', { async: true }),
+  appId: Ember.computed.alias('scmRepo.name'),
```
Comment (tkyi) [inline app/pipeline/model.js]: Cool! 👍 Much cleaner than needing to use the `get()` on the `Ember.Object.create(this.get('scmRepo'));`

File: app/pipeline/serializer.js [hunk 2]
```diff
@@ -8,6 +8,6 @@ export default DS.RESTSerializer.extend({
    * @method serializeIntoHash
    */
   serializeIntoHash(hash, typeClass, snapshot) {
-    return Ember.merge(hash, { checkoutUrl: snapshot.attr('scmUrl') });
+    return Ember.merge(hash, { checkoutUrl: snapshot.attr('checkoutUrl') });
```
Comment (tkyi) [inline app/pipeline/serializer.js]: You'll need to update this file to use the new checkoutUrl too: `{ checkoutUrl }`
https://github.com/screwdriver-cd/ui/blob/master/app/create/controller.js#L10

Comment (petey) [inline app/pipeline/serializer.js]: good catch

File: app/event/model.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    get() {
+      const duration = Date.now() - this.get('createTime').getTime();
+
+      return `${humanizeDuration(duration, { round: true, largest: 1 })} ago`;
+    }
+  }),
+  creator: DS.attr(),
+  duration: Ember.computed('builds', {
+    get() {
+      return this.get('builds').reduce((val = 0, item) => val + item.get('totalDurationMS'));
```
Comment (minz1027) [inline app/event/model.js]: Nice!

File: app/helpers/index-of.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import Ember from 'ember';
+
+/**
+ * get an element of the array at the specified index
+ * @method indexOf
+ * @param  {Array} params   [contextArr, index]
+ * @return {Mixed}
+ */
+export function indexOf(params) {
```
Comment (minz1027) [inline app/helpers/index-of.js]: Are we using it any where?

Comment (petey) [inline app/helpers/index-of.js]: https://github.com/screwdriver-cd/ui/blob/468440e8fa3cf0fefb25ca0420e025285bc53732/app/components/pipeline-event-view/template.hbs#L17

**Comment by minz1027** [comment]
Resolves https://github.com/screwdriver-cd/screwdriver/issues/354

File: app/helpers/index-of.js (hunk 4 above)
Comment (minz1027) [inline app/helpers/index-of.js]: Cool!

**Pull request merged**